### PR TITLE
Set-DbaLogin - add a param so that check_expiration can be set

### LINKED ISSUE #7090: Set-DbaLogin - add a param for check_expiration (SMO PasswordExpirationEnabled)
### Report
Set-DbaLogin - add a param for check_expiration (SMO PasswordExpirationEnabled)

While working on #7004 it was noticed that the caller does not have the ability to set the PasswordExpirationEnabled property.

Suggested code changes are here:

**New param**
https://github.com/sqlcollaborative/dbatools/pull/7087/files/7127b1c70d9ad43de2d5fa412ab4e9ea485f57f5#diff-419523c18ac6bcaa3c3134c4b1996958ae756dbd7a59abec0193bfb8dc804381R179

**Logic for password expiration param**
https://github.com/sqlcollaborative/dbatools/pull/7087/files/7127b1c70d9ad43de2d5fa412ab4e9ea485f57f5#diff-419523c18ac6bcaa3c3134c4b1996958ae756dbd7a59abec0193bfb8dc804381R349

https://github.com/sqlcollaborative/dbatools/pull/7087/files/7127b1c70d9ad43de2d5fa412ab4e9ea485f57f5#diff-419523c18ac6bcaa3c3134c4b1996958ae756dbd7a59abec0193bfb8dc804381R420

#### Host used
- [x] powershell.exe
- [ ] ISE
- [ ] VS Code
- [ ] Other (please specify)

#### Steps to Reproduce
Set-DbaLogin -SqlInstance $server -Login SQLUser1 -PasswordExpirationEnabled

#### Expected Behavior
The caller should be able to set the PasswordExpirationEnabled login property.

#### Actual Behavior
Unable to set the PasswordExpirationEnabled property on the login object.

### Environmental information
PowerShell Version : 5.1.18362.1171
dbatools latest installed : latest from the development branch
Culture of OS : en-US

SQL Server:
Microsoft SQL Server 2019 (RTM-CU8-GDR) (KB4583459) - 15.0.4083.2 (X64) Nov 2 2020 18:35:09 Copyright (C) 2019 Microsoft Corporation Developer Edition (64-bit) on Windows 10 Pro 10.0 (Build 18363: ) (Hypervisor)

us_english

### PR DESCRIPTION
**Pull request by lancasteradam**
Set-DbaLogin - add a param so that check_expiration can be set

- Added a new param so that the check_expiration for the login can be set
- Added a new integration test

## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #7090 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
 
Adding the option to set the check_expiration for the login.

### Commands to test

See the integration test for the PasswordExpirationEnabled param.

### CODE REVIEW

**Comment by lancasteradam** [comment]
@wsmelton, looks like you are doing a major restructuring of the repo. This PR can wait or if necessary I can resubmit it once the repo is restructured.

File: functions/Set-DbaLogin.ps1 [hunk 1]
```diff
@@ -50,7 +50,10 @@ function Set-DbaLogin {
         Grant access to SQL Server
 
     .PARAMETER PasswordPolicyEnforced
-        Should the password policy be enforced.
+        Should the password policy be enforced (check_policy).
```
Comment (wsmelton) [inline functions/Set-DbaLogin.ps1:53]:
```suggestion
        Enable the password policy on the login (check_policy = ON).
```

Saying "Should" just sounds too much like a Pester tests 🤣

File: functions/Set-DbaLogin.ps1 [hunk 2]
```diff
@@ -50,7 +50,10 @@ function Set-DbaLogin {
...
+
+    .PARAMETER PasswordExpirationEnabled
+        Should the password expiration check be enforced (check_expiration). In order to enable this option the PasswordPolicyEnforced (check_policy) must also be enabled for the login.
```
Comment (wsmelton) [inline functions/Set-DbaLogin.ps1:56]: Do we need to validate this before trying to enable it? Or depend on the error that SQL Server will return?

File: functions/Set-DbaLogin.ps1 (hunk 1 above)
Comment (lancasteradam) [reply functions/Set-DbaLogin.ps1:53]: yeah, I agree

File: functions/Set-DbaLogin.ps1 (hunk 2 above)
Comment (lancasteradam) [reply functions/Set-DbaLogin.ps1:56]: Yeah, good catch. I was originally thinking to allow the SQL error to propagate back to the caller, but that would mean the alter() would generate an exception and stop processing other logins. I've added in some validation and an error message to help the caller know what they can do to correct the error.

**Comment by lancasteradam** [comment]
@wsmelton, friendly reminder on this PR.

**Comment by potatoqualitee** [comment]
@lancasteradam - my sincerest apology for the delays. this part of the lockdown is positively crushing and I've been running from larger GitHub projects. I believe I've got things a bit more under control now and will work to merge your 🔥 PRs faster. Thanks again for all of the incredible enhancements.

**Review by jpomfret — APPROVED** [review]
Looks good!

**Comment by lancasteradam** [comment]
Thank you @potatoqualitee and @jpomfret.

**Pull request merged**